src/controller/python/chip/ChipCoreBluetoothMgr.py

```diff
@@ -220,8 +243,12 @@ def shouldLoop(self, should_tuple):
             if not time_expired:
                 result = True
 
-            for peripheral in self.peripheral_list:
-                if should_tuple[3] and str(peripheral._.name) == should_tuple[3]:
+            for peripheral in self.peripheral_adv_list:
+                if should_tuple[3] and str(peripheral.peripheral._.name) == should_tuple[3]:
+                    result = False
+                    break
+                devIdInfo = peripheral.getPeripheralDevIdInfo()
+                if devIdInfo and should_tuple[3] and str(devIdInfo.discriminator) == should_tuple[3]:
                     result = False
```

Comment:
`return False` (without single break) is much more pythonic. 

code shoulld likely look like:

```python

if time_expired:
  return False

status = should_tuple[0]

if status == "ready":
    return (not self.ready_condition)
elif status == "scan":
   # .... for loops for name checks
   return True
elif status == "connect":
   return (not self.loop_condition)
.....

return False
```

Reply:
Aggree